src/librustc_mir/borrow_check/nll/constraints/graph.rs

```diff
@@ -129,7 +161,21 @@ impl<'s, D: ConstraintGraphDirecton> Iterator for Edges<'s, D> {
     fn next(&mut self) -> Option<Self::Item> {
         if let Some(p) = self.pointer {
             self.pointer = self.graph.next_constraints[p];
+
             Some(self.constraints[p])
+        } else if let Some(next_static_idx) = self.next_static_idx {
+            self.next_static_idx =
+                if next_static_idx == (self.graph.first_constraints.len() - 1) {
+                    None
+                } else {
+                    Some(next_static_idx + 1)
+                };
+
+            Some(OutlivesConstraint {
+                sup: self.static_region,
+                sub: next_static_idx.into(),
```

Comment:
@nikomatsakis  I was looking at this again last night and had a thought. Should this line actually be something like this?

```
sub: self.constraints[self.graph.first_constraints[next_static_idx]].sub,
```

Probably not since the compiler ICEs when bootstrapping now but perhaps I messed up something else 😄